packages/grid/x-data-grid/src/hooks/features/clipboard/useGridClipboard.ts

```diff
@@ -81,16 +84,36 @@ export const useGridClipboard = (apiRef: React.MutableRefObject<GridPrivateApiCo
         return;
       }
 
-      apiRef.current.unstable_copySelectedRowsToClipboard();
+      let textToCopy = '';
+      const selectedRows = apiRef.current.getSelectedRows();
+      if (selectedRows.size > 0) {
+        textToCopy = apiRef.current.getDataAsCsv({
+          includeHeaders: false,
```

Comment:
One use-case could be to include headers for copy and paste, do you think it's worth considering introducing props like `unstable_includeHeadersDuringExport` and `unstable_includeHeadersDuringImport`? (perhaps in a follow-up PR)

Reply:
> `unstable_includeHeadersDuringExport`

We removed the keyboard shortcut for "copy with headers" in https://github.com/mui/mui-x/issues/5618
There were no requests to support this - see https://github.com/mui/mui-x/issues?q=is%3Aissue+clipboard+headers+
I would prefer to not add it until it's requested :)
Also, I'm not sure how we should handle a paste operation that includes column headers.

> `unstable_includeHeadersDuringImport`

I'm not sure how this prop would impact the clipboard import behavior, can you elaborate?

Reply:
You are right, defining and implementing how a paste operation work on headers is complex, and may be followed up when (if) requested.